File: Java/1-Arrays/Intervals/Non_Overlapping_Intervals.java
class Non_Overlapping_Intervals {
    public int eraseOverlapIntervals(int[][] intervals) {
        
        if (intervals.length == 0) {
            return 0;
        }
        
        Arrays.sort(intervals, (o1, o2) -> Integer.compare(o1[1], o2[1]));
        
        int prevEnd = intervals[0][1];
        int count = 0;
        for(int i = 1; i < intervals.length; i++) {
            if (intervals[i][0] < prevEnd) {
                count++;
            } else {
                prevEnd = intervals[i][1];
            }
        }
        
        return count;
    }
}

File: Python/0_______ARRAY_______/Intervals/Non_Overlapping_Intervals.py
from ast import List

class Solution:
    # We want to erase the intervals so that minimum amount of intervals will be conflicting. 
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        # first of all, list the intervals
        intervals.sort(key=lambda x: x[0])
        
        count = 0
        prevEnd = intervals[0][1]
        
        for interval in intervals[1: ]:
            if prevEnd <= interval[0]:
                prevEnd = interval[1]
            else:
                count += 1
                prevEnd = min(prevEnd, interval[1])
        return count